### ISSUE TITLE
Wrong Event Recurring count

### ISSUE THREAD

**Issue by ABDows** [DHTMLX/scheduler-helper-php#10]
Hi,
I created a recurring event repeated 5 times then I delete one occurence but S-helper return 6 recurrences.
It should return (5-1) events !? or not ? 

I think there are 2 problème : 
1-  for rec_type like day_1___#4 : it returned 5 occurences ( should be 4 occurences )
2- the deleted recurrences are not detected.

**Comment by ABDows** [comment]
Hi, I still have the same problem I tried many solution without valide results.

Below my code :
PHP CODE :

``` php
public function index()    {
        $helper = new Helper(
            array(
                "dbsm" => "mysql", // optional, "mysql" by default
                "host" => "localhost", // optional, "localhost" by default
                "db_name" => "msa_new_db",
                "user" => "root",
                "password" => "",
                "table_name" => "time_slots" // name of the table that contains data of recurring events
            )
        );

        $helper->setFieldsNames(array(
            helper::FLD_ID => "time_slot_id",
        ));

        $red = $helper->getData("2015-01-01 09:00:00", "2020-01-02 07:00:00");

        var_dump($red);
    }
```

SQL TABLE :

```
CREATE TABLE `time_slots` (
  `time_slot_id` int(11) NOT NULL AUTO_INCREMENT,
  `block_online` tinyint(1) DEFAULT NULL,
  `agenda_id` int(11) DEFAULT NULL,
  `title` varchar(50) COLLATE utf8_unicode_ci DEFAULT NULL,
  `description` text COLLATE utf8_unicode_ci,
  `start_date` datetime DEFAULT NULL,
  `end_date` datetime DEFAULT NULL,
  `event_type` char(1) COLLATE utf8_unicode_ci DEFAULT NULL,
  `rec_type` varchar(30) COLLATE utf8_unicode_ci DEFAULT NULL,
  `event_pid` int(11) DEFAULT NULL,
  `event_length` int(11) DEFAULT '0',
  `color` varchar(8) COLLATE utf8_unicode_ci DEFAULT NULL,
  PRIMARY KEY (`time_slot_id`),
  KEY `FK_Reference_33` (`agenda_id`),
  CONSTRAINT `FK_Reference_33` FOREIGN KEY (`agenda_id`) REFERENCES `agendas` (`agenda_id`)
) ENGINE=InnoDB AUTO_INCREMENT=35 DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```

DB ROWS: ![Alt Text](http://forum.dhtmlx.com/download/file.php?id=9599)

When I called index Function the result go wrong : 
Result : 

``` PHP
array (size=7)
  0 => 
    array (size=12)
      'time_slot_id' => string '33' (length=2)
      'block_online' => string '0' (length=1)
      'agenda_id' => string '1' (length=1)
      'title' => string 'AAA' (length=3)
      'description' => string 'AAA' (length=3)
      'start_date' => string '2015-10-07 07:40:00' (length=19)
      'end_date' => string '2015-10-07 10:45:00' (length=19)
      'event_type' => string 'p' (length=1)
      'rec_type' => string 'day_1___#6' (length=10)
      'event_pid' => string '0' (length=1)
      'event_length' => string '11100' (length=5)
      'color' => string '#90EE90' (length=7)
  1 => 
    array (size=12)
      'time_slot_id' => string '33' (length=2)
      'block_online' => string '0' (length=1)
      'agenda_id' => string '1' (length=1)
      'title' => string 'AAA' (length=3)
      'description' => string 'AAA' (length=3)
      'start_date' => string '2015-10-08 07:40:00' (length=19)
      'end_date' => string '2015-10-08 10:45:00' (length=19)
      'event_type' => string 'p' (length=1)
      'rec_type' => string 'day_1___#6' (length=10)
      'event_pid' => string '0' (length=1)
      'event_length' => string '11100' (length=5)
      'color' => string '#90EE90' (length=7)
  2 => 
    array (size=12)
      'time_slot_id' => string '33' (length=2)
      'block_online' => string '0' (length=1)
      'agenda_id' => string '1' (length=1)
      'title' => string 'AAA' (length=3)
      'description' => string 'AAA' (length=3)
      'start_date' => string '2015-10-09 07:40:00' (length=19)
      'end_date' => string '2015-10-09 10:45:00' (length=19)
      'event_type' => string 'p' (length=1)
      'rec_type' => string 'day_1___#6' (length=10)
      'event_pid' => string '0' (length=1)
      'event_length' => string '11100' (length=5)
      'color' => string '#90EE90' (length=7)
  3 => 
    array (size=12)
      'time_slot_id' => string '33' (length=2)
      'block_online' => string '0' (length=1)
      'agenda_id' => string '1' (length=1)
      'title' => string 'AAA' (length=3)
      'description' => string 'AAA' (length=3)
      'start_date' => string '2015-10-10 07:40:00' (length=19)
      'end_date' => string '2015-10-10 10:45:00' (length=19)
      'event_type' => string 'p' (length=1)
      'rec_type' => string 'day_1___#6' (length=10)
      'event_pid' => string '0' (length=1)
      'event_length' => string '11100' (length=5)
      'color' => string '#90EE90' (length=7)
  4 => 
    array (size=12)
      'time_slot_id' => string '33' (length=2)
      'block_online' => string '0' (length=1)
      'agenda_id' => string '1' (length=1)
      'title' => string 'AAA' (length=3)
      'description' => string 'AAA' (length=3)
      'start_date' => string '2015-10-11 07:40:00' (length=19)
      'end_date' => string '2015-10-11 10:45:00' (length=19)
      'event_type' => string 'p' (length=1)
      'rec_type' => string 'day_1___#6' (length=10)
      'event_pid' => string '0' (length=1)
      'event_length' => string '11100' (length=5)
      'color' => string '#90EE90' (length=7)
  5 => 
    array (size=12)
      'time_slot_id' => string '33' (length=2)
      'block_online' => string '0' (length=1)
      'agenda_id' => string '1' (length=1)
      'title' => string 'AAA' (length=3)
      'description' => string 'AAA' (length=3)
      'start_date' => string '2015-10-12 07:40:00' (length=19)
      'end_date' => string '2015-10-12 10:45:00' (length=19)
      'event_type' => string 'p' (length=1)
      'rec_type' => string 'day_1___#6' (length=10)
      'event_pid' => string '0' (length=1)
      'event_length' => string '11100' (length=5)
      'color' => string '#90EE90' (length=7)
  6 => 
    array (size=12)
      'time_slot_id' => string '33' (length=2)
      'block_online' => string '0' (length=1)
      'agenda_id' => string '1' (length=1)
      'title' => string 'AAA' (length=3)
      'description' => string 'AAA' (length=3)
      'start_date' => string '2015-10-13 07:40:00' (length=19)
      'end_date' => string '2015-10-13 10:45:00' (length=19)
      'event_type' => string 'p' (length=1)
      'rec_type' => string 'day_1___#6' (length=10)
      'event_pid' => string '0' (length=1)
      'event_length' => string '11100' (length=5)
      'color' => string '#90EE90' (length=7)
```

**Comment by ABDows** [comment]
I Make Some change To get Helper Works , I don't know If is the correct way but the result is correct for me.
I replaced This line 

``` PHP
$events[$eventParentId][$eventLength] = $eventData; 
```

By

``` PHP
$events[$eventParentId][$this->getDateTimestamp($eventData[$this->getStartDateFieldName()])] = $eventData;      
```

In 

``` PHP
_getRecurringEventsExceptionsByInterval Function
```

What do you think ?

**Comment by AlexKlimenkov** [comment]
Fixed in v1.1.0
https://github.com/DHTMLX/scheduler-helper-php/releases/tag/1.1.0
